zebra-rpc/qa/rpc-tests/feature_backup_non_finalized_state.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+from decimal import Decimal
+from test_framework.util import assert_equal, start_nodes
+from test_framework.test_framework import BitcoinTestFramework
+import time
+
+# Test that Zebra can backup and restore non finalized state
+class BackupNonFinalized(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 1
+        self.cache_behavior = 'clean'
+
+    def setup_network(self):
+        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir, [None])
+
+    def run_test(self):
+        self.nodes[0].generate(10)
+        # Wait for `MIN_DURATION_BETWEEN_BACKUP_UPDATES`, 5 secs in zebra
+        time.sleep(5)
+
+        blocks = self.nodes[0].getblockchaininfo()['blocks']
+        assert_equal(blocks, 10)
+
+        # Stop the node
+        self.nodes[0].stop()
+        time.sleep(1)
+
+        self.nodes = start_nodes(self.num_nodes, self.options.tmpdir, [None])
+
+        # Good, the node has recovered the non finalized state
+        blocks = self.nodes[0].getblockchaininfo()['blocks']
+        assert_equal(blocks, 10)
+
+        # Generate more blocks and make sure the blockchain is still working
+        self.nodes[0].generate(1)
+        time.sleep(5)
```

Comment:
Sleep here is probably unnecessary.

Reply:
Right, 1 seconds is needed after generation if we have a wallet so it can sync with Zebra. No wallet here so no need to wait. Removed sleep's at https://github.com/ZcashFoundation/zebra/pull/9896/commits/9ee495821681c8b7e0dd1c1f365b3b25ef00677b